DragaliaAPI.Test/Integration/Dragalia/DragonControllerTest.cs

```diff
@@ -0,0 +1,578 @@
+        {
+            returnDragon.attack_plus_count.Should().Be(dbDragonSacrifice.AttackPlusCount);
+            returnDragon.hp_plus_count.Should().Be(dbDragonSacrifice.HpPlusCount);
+        }
+    }
+
+    [Fact]
+    public async Task DragonBuildUp_ReturnsDragonDataWithAugments()
+    {
+        ApiContext context = fixture.Services.GetRequiredService<ApiContext>();
+
+        DbPlayerDragonData dbDragon = context.PlayerDragonData
+            .Add(DbPlayerDragonDataFactory.Create(DeviceAccountIdConst, Dragons.Liger))
+            .Entity;
+
+        await context.SaveChangesAsync();
+
+        DragonBuildupRequest request = new DragonBuildupRequest()
+        {
+            base_dragon_key_id = (ulong)dbDragon.DragonKeyId,
+            grow_material_list = new List<GrowMaterialList>()
+            {
+                new GrowMaterialList()
+                {
+                    type = EntityTypes.Material,
+                    id = (int)Materials.AmplifyingDragonscale,
+                    quantity = 25
+                }
+            }
+        };
+
+        DragonBuildupData? response = (
+            await client.PostMsgpack<DragonBuildupData>("dragon/buildup", request)
+        ).data;
+
+        response.Should().NotBeNull();
+        response.update_data_list.Should().NotBeNull();
+        response.update_data_list.dragon_list.Should().NotBeNullOrEmpty();
+        DragonList returnDragon = response.update_data_list.dragon_list
+            .Where(x => (long)x.dragon_key_id == dbDragon.DragonKeyId)
+            .First();
+        returnDragon.attack_plus_count.Should().Be(25);
+    }
+
+    [Fact]
+    public async Task DragonResetPlusCount_ResetsAugments()
+    {
+        ApiContext context = fixture.Services.GetRequiredService<ApiContext>();
+
+        DbPlayerDragonData dragon = DbPlayerDragonDataFactory.Create(
+            DeviceAccountIdConst,
+            Dragons.Maritimus
+        );
+        dragon.AttackPlusCount = 50;
+        dragon = context.PlayerDragonData.Add(dragon).Entity;
+
+        await context.SaveChangesAsync();
+
+        context.ChangeTracker.Clear();
+        DbPlayerUserData userData = await context.PlayerUserData
+            .Where(x => x.DeviceAccountId == DeviceAccountIdConst)
+            .FirstAsync();
+
+        long startCoin = userData.Coin;
+
+        DragonResetPlusCountRequest request = new DragonResetPlusCountRequest()
+        {
+            dragon_key_id = (ulong)dragon.DragonKeyId,
+            plus_count_type = (int)UpgradeEnhanceTypes.AtkPlus
+        };
+
+        DragonSetLockData? response = (
+            await client.PostMsgpack<DragonSetLockData>("dragon/reset_plus_count", request)
+        ).data;
+
+        response.Should().NotBeNull();
+        response.update_data_list.Should().NotBeNull();
+        response.update_data_list.dragon_list.Should().NotBeNullOrEmpty();
+        DragonList returnDragon = response.update_data_list.dragon_list
```

Comment:
You don't have to fix this, but FYI the preceding 3 assertions are not needed because this statement would throw an NullReference/InvalidOperation exception if they were untrue and the test would fail anyway